main.go

```diff
@@ -156,13 +156,32 @@ func main() {
 		"chs": {"Change signal of x", 1, false, func(a []float64) (float64, error) { return a[0] * -1, nil }},
 		"inv": {"Invert x (1/x)", 1, false, func(a []float64) (float64, error) { return 1 / a[0], nil }},
 		"^":   {"Raise y to the power of x", 2, false, func(a []float64) (float64, error) { return math.Pow(a[0], a[1]), nil }},
-		"%":   {"Calculates y modulo x", 2, false, func(a []float64) (float64, error) { return math.Mod(a[0], a[1]), nil }},
-		"pct": {"Calculate x% of y", 2, false, func(a []float64) (float64, error) { return a[0] * a[1] / 100, nil }},
+		"mod":   {"Calculates y modulo x", 2, false, func(a []float64) (float64, error) { return math.Mod(a[0], a[1]), nil }},
+		"%": {"Calculate x% of y", 2, false, func(a []float64) (float64, error) { return a[0] * a[1] / 100, nil }},
+
+        "fac": {"Calculate factorial of x", 1, false, func(a []float64) (float64, error) {
+            var x = uint64(a[0])
+            if (x <= 0) {
+                return 0, errors.New("fact requires a positive number")
+            }
+            var fact = uint64(1)
+            for ix := uint64(1); ix <= x; ix++ {
+			    fact *= ix
```

Comment:
Indentation issues? Please run gofmt on the entire source.

Reply:
done.